batch/batches/Storage/Periodic/KAsyncStoragePeriodicExport.class.php

```diff
@@ -0,0 +1,267 @@
+<?php
+
+/**
+ * process remote storage files and export using file handlers as required
+ *
+ * @package Scheduler
+ * @subpackage Storage
+ */
+class KAsyncStoragePeriodicExport extends KPeriodicWorker
+{
+	const MAX_EXECUTION_TIME = 3600;        // can be exceeded by one file sync
+	const IDLE_SLEEP_INTERVAL = 10;
+	const MAX_COUNT = 'maxCount';
+	const MAX_SIZE = 'maxSize';
+	const STORAGE_PROFILE_IDS = 'profileIdsIn';
+	const RESPONSE_PROFILE_FIELDS = 'id,originalId,fileSize,fileRoot,filePath,isDir,srcPath,srcEncKey';
+
+	protected $storageProfiles;
+	protected $currentIndex;
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::STORAGE_PERIODIC_EXPORT;
+	}
+
+	/* (non-PHPdoc)
+	 * @see KBatchBase::run()
+	*/
+	public function run($jobs = null)
+	{
+		$filter = $this->getFilter();
+		$maxCount = $this->getAdditionalParams(self::MAX_COUNT);
+		$maxSize = $this->getAdditionalParams(self::MAX_SIZE);
+		$this->storageProfileIdsArray = explode(',', $this->getAdditionalParams(self::STORAGE_PROFILE_IDS));
+		if(!$this->storageProfileIdsArray)
+		{
+			KalturaLog::debug('Storage Profile Ids are missing from config');
+			return;
+		}
+		$this->currentIndex = 0;
+		$responseProfile = $this->initResponseProfile();
+		$timeLimit = time() + self::MAX_EXECUTION_TIME;
+
+		while (time() < $timeLimit)
+		{
+			$storageProfile = $this->getStorageProfile();
+			if(!$storageProfile)
+			{
+				continue;
+			}
+
+			self::$kClient->setResponseProfile($responseProfile);
+			$lockResult = self::$kClient->storageProfile->lockPendingFileSyncs($filter, $this->getId(), $storageProfile->id, $maxCount, $maxSize);
+			if (!$lockResult->fileSyncs)
+			{
+				sleep(self::IDLE_SLEEP_INTERVAL);
+				continue;
+			}
+
+			$this->exportFileSyncs($lockResult->fileSyncs, $storageProfile);
+
+			if (!$lockResult->limitReached)
+			{
+				sleep(self::IDLE_SLEEP_INTERVAL);
+			}
+		}
+	}
+
+	protected function getFilter()
+	{
+		$filter = new KalturaFileSyncFilter();
+		if (KBatchBase::$taskConfig->filter)
+		{
+			// copy the attributes since KBatchBase::$taskConfig->filter is of type KalturaBatchJobFilter
+			foreach (KBatchBase::$taskConfig->filter as $attr => $value)
+			{
+				$filter->$attr = $value;
+			}
+		}
+		return $filter;
+	}
+
+	protected function export($storageProfile, $fileSync)
+	{
+		$data = $this->createExportData($storageProfile, $fileSync);
+		$engine = KExportEngine::getInstance($storageProfile->protocol, $storageProfile->partnerId, $data);
+		if (!$engine)
+		{
+			KalturaLog::debug('Engine not found');
+			return false;
+		}
+
+		$exportResult = $engine->export();
+		if(!$exportResult)
+		{
+			KalturaLog::debug('File export failed');
+		}
+		return $exportResult;
+	}
+
+
+	protected function createExportData($storageProfile, $fileSync)
```

Comment:
This priodic exported must not have anything dedicated to S3 or any other storage system. Try move the specific type handling to different file, most likely the engine code is a good place.